### ISSUE TITLE
Problems saving twitter threads

### ISSUE THREAD

**Issue by idlethread** [patmigliaccio/logseq-archive-webpage#5]
{{renderer archive https://twitter.com/BrianFeroldi/status/1690699697458122754}} doesn't seem to produce a valid archive.

This is likely because of twitter make it hard to scrape a thread, possibly. But reporting here, to see if this is expected behaviour.

Thanks for working on this plugin though from the [conversation thread](https://discuss.logseq.com/t/any-way-to-archive-external-urls-transparently/21354) in the logseq discussion forum.

**Comment by patmigliaccio** [comment]
You're right on. Twitter has put blockers in place to prevent scraping. Even the Internet Archive is having trouble with this [link](https://web.archive.org/web/20231207032645/https://twitter.com/BrianFeroldi/status/1690699697458122754/). But Logseq should be able to embed Twitter links with `{{twitter [url]}}`.

On our end, the local archival process is limited because Logseq plugins are sandboxed to only support client-side JavaScript. Since we can only use what we can run in the browser, we're unable to spin up a more sophisticated renderer for pages.

I have #1 open to potentially setup an API that'll render the pages remotely. But that defeats the purpose of archiving locally.

Sorry about this one and thanks for the feedback.

**Comment by davidak** [comment]
> This is likely because of twitter make it hard to scrape a thread

I noticed that i can't even see the whole thread anymore without being logged in, just the specific linked post. So Twitter isn't really a public source anymore.

You can see threads and comments with [Nitter](https://nitter.net/about), an alternative open source Twitter front-end focused on privacy. Consider supporting the creator if you see value in it.

Example: https://nitter.net/BrianFeroldi/status/1690699697458122754

Just replace the domain `twitter.com` with `nitter.net` in a URL to use it. You can archive that URL then.

Example: https://web.archive.org/web/20240109171846/https://nitter.net/BrianFeroldi/status/1690699697458122754

There is also [Thread Reader](https://threadreaderapp.com/), but you have to be logged in to Twitter to ask their bot to archive a thread.

Example:

https://threadreaderapp.com/thread/1690699697458122754.html
https://web.archive.org/web/20240109172431/https://threadreaderapp.com/thread/1690699697458122754.html

I'm not sure if these services would be a trusted source in academic context since these services could technically change the content. You might have to save the Twitter page directly while being logged in, e.g. with some browser extension.
